### PR TITLE
refactor: use Promise.all to await for independent promises

### PR DESCRIPTION
**Pull request by pepakriz**

### CODE REVIEW

File: tests/issues/GH228.test.ts [hunk 1]
```diff
@@ -69,7 +69,7 @@ describe('GH issue 228', () => {
       populate: true,
     });
     expect(mock.mock.calls[0][0]).toMatch('select `e0`.* from `a` as `e0` order by `e0`.`type_id` asc');
-    expect(mock.mock.calls[1][0]).toMatch('select `e0`.* from `b` as `e0` where `e0`.`id` in (?)');
+    expect(mock.mock.calls[2][0]).toMatch('select `e0`.* from `b` as `e0` where `e0`.`id` in (?)');
```
Comment (pepakriz) [inline tests/issues/GH228.test.ts]: This test depends on the execution order :(

Comment (B4nan) [reply tests/issues/GH228.test.ts]: I guess we could reorder the `calls` array then? Or check its length (to be sure we do not fire more queries than expected) and check for particular queries via sth like `includes`. 

```typescript
const queries = mock.mock.calls.map(c => c[0]);
expect(queries).toHaveLength(3);
expect(queries).toContain('... sql1 ...');
expect(queries).toContain('... sql2 ...');
```

Comment (pepakriz) [reply tests/issues/GH228.test.ts]: Thanks for suggestions. I've used sort fn.

**Comment by B4nan** [comment]
Hmm looks like its stuck again, what ever, thanks!

**Comment by pepakriz** [comment]
@B4nan It was stuck because of this issue: https://github.com/mikro-orm/mikro-orm/pull/387
You should wait for the "green" pipeline :)

**Pull request merged**